### PR TITLE
Vet clinic database: add "join table" for visits

### PR DESCRIPTION
**Pull request by Yidnekachew-cmd**
### In this branch, I have added the following; 

- [ ] Create a table named vets with the following columns:
    - id: integer (set it as autoincremented PRIMARY KEY)
    - name: string
    - age: integer
    - date_of_graduation: date
- [ ] Create a "join table" called specializations to handle the relationship between the tables `species` and `vets`.
- [ ] Create a "join table" called visits to handle the relationship between the tables `animals` and `vets`.
- [ ] Insert the following data for vets:
     - Vet William Tatcher is 45 years old and graduated Apr 23rd, 2000.
    - Vet Maisy Smith is 26 years old and graduated Jan 17th, 2019.
    - Vet Stephanie Mendez is 64 years old and graduated May 4th, 1981.
    - Vet Jack Harkness is 38 years old and graduated Jun 8th, 2008.
- [ ] Insert the following data for specialties:
     - Vet William Tatcher is specialized in Pokemon.
    - Vet Stephanie Mendez is specialized in Digimon and Pokemon.
    - Vet Jack Harkness is specialized in Digimon.
- [ ] Insert the following data for visits:
    - Agumon visited William Tatcher on May 24th, 2020.
    - Agumon visited Stephanie Mendez on Jul 22th, 2020.
   - Gabumon visited Jack Harkness on Feb 2nd, 2021.
   - Pikachu visited Maisy Smith on Jan 5th, 2020.
   - Pikachu visited Maisy Smith on Mar 8th, 2020.
   - Pikachu visited Maisy Smith on May 14th, 2020.
   - Devimon visited Stephanie Mendez on May 4th, 2021.
   - Charmander visited Jack Harkness on Feb 24th, 2021.
   - Plantmon visited Maisy Smith on Dec 21st, 2019.
   - Plantmon visited William Tatcher on Aug 10th, 2020.
   - Plantmon visited Maisy Smith on Apr 7th, 2021.
   - Squirtle visited Stephanie Mendez on Sep 29th, 2019.
   - Angemon visited Jack Harkness on Oct 3rd, 2020.
   - Angemon visited Jack Harkness on Nov 4th, 2020.
   - Boarmon visited Maisy Smith on Jan 24th, 2019.
   - Boarmon visited Maisy Smith on May 15th, 2019.
   - Boarmon visited Maisy Smith on Feb 27th, 2020.
   - Boarmon visited Maisy Smith on Aug 3rd, 2020.
   - Blossom visited Stephanie Mendez on May 24th, 2020.
  - Blossom visited William Tatcher on Jan 11th, 2021.
- [ ] Write queries to answer the following:
    - Who was the last animal seen by William Tatcher?
    - How many different animals did Stephanie Mendez see?
    - List all vets and their specialties, including vets with no specialties.
    - List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
    - What animal has the most visits to vets?
    - Who was Maisy Smith's first visit?
    - Details for most recent visit: animal information, vet information, and date of visit.
    - How many visits were with a vet that did not specialize in that animal's species?
    - What specialty should Maisy Smith consider getting? Look for the species she gets the most.
- [ ] Screenshot of the results of your queries.
Screenshot from 2023-05-11 18-03-27
Screenshot from 2023-05-11 18-03-54
Screenshot from 2023-05-11 18-04-16
Screenshot from 2023-05-11 18-04-34
Screenshot from 2023-05-11 18-05-02
Screenshot from 2023-05-11 18-05-28
Screenshot from 2023-05-11 18-05-50
Screenshot from 2023-05-11 18-06-15
Screenshot from 2023-05-11 18-06-35

- [ ] Populate the specializations table with the given data.
image

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -64,4 +64,26 @@ UPDATE animals SET owner_id = 1 WHERE name = 'Agumon';
...
+--Insert data to vets table:
+
+INSERT INTO vets (name, age, date_of_graduation)
+VALUES
+  ('William Tatcher', 45, '2000-04-23'),
+  ('Maisy Smith', 26, '2019-01-17'),
+  ('Stephanie Mendez', 64, '1981-05-04'),
+  ('Jack Harkness', 38, '2008-06-08');
+
+--Insert data to specializations table:
```
Comment (amon-cofie) [inline data.sql:78]: You did great work creating the specializations table 👍. However, I think you forgot to populate the table with the required data. I suggest that you populate the specializations table with the data as required in the project requirements: 
image

The code should probably look something like this 😉: 
```sh
 INSERT INTO specializations (species_id, vet_id)
 VALUES ((SELECT id FROM vets WHERE name = 'William Tatcher'), (SELECT id FROM species WHERE name = 'Pokemon'));

 INSERT INTO specializations (species_id, vet_id)
 VALUES ((SELECT id FROM vets WHERE name = 'Stephanie Mendez'), (SELECT id FROM species WHERE name = 'Pokemon'));

 INSERT INTO specializations (species_id, vet_id)
 VALUES ((SELECT id FROM vets WHERE name = 'Stephanie Mendez'), (SELECT id FROM species WHERE name = 'Digimon'));

 INSERT INTO specializations (species_id, vet_id)
 VALUES ((SELECT id FROM vets WHERE name = 'Jack Harkness'), (SELECT id FROM species WHERE name = 'Digimon'));

```

**Review by amon-cofie — CHANGES_REQUESTED** [review]
Hi @Yidnekachew-cmd , 👋 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there! 💪 



## Highlights

🚀 You have a professional Readme file
🚀 You used Gitflow properly
🚀 You annotated your code very well and made it very easy to read

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Shedrack-Sunday — APPROVED** [review]
Hi @Yidnekachew-cmd  

Congratulations! 🎉 Your project is complete and ready to be merged! 🚀

image


## Highlights
- Good use of SQL queries✔️ 

## Optional Suggestions 💡
While these comments with the [OPTIONAL] prefix won't prevent the approval of this pull request, I strongly recommend taking them into account as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really enhance your application.
Cheers, and Happy coding!👏👏👏

If you have any questions or need clarification on anything, feel free to leave a comment in the pull request thread, and don't forget to tag me so I'll receive the notification.

As per the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project. If you feel that the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by Yidnekachew-cmd** [comment]
Hi @Shedrack-Sunday, Thank you!  :hugs:

**Pull request merged**